crates/astria-conductor/src/executor/client.rs

```diff
@@ -168,3 +216,172 @@ fn block_identifier(number: u32) -> raw::BlockIdentifier {
         identifier: Some(raw::block_identifier::Identifier::BlockNumber(number)),
     }
 }
+
+struct OnRetry {
+    parent: Span,
+}
+
+impl tryhard::OnRetry<tonic::Status> for OnRetry {
+    type Future = futures::future::Ready<()>;
+
+    fn on_retry(
+        &mut self,
+        attempt: u32,
+        next_delay: Option<Duration>,
+        previous_error: &tonic::Status,
+    ) -> Self::Future {
+        let wait_duration = next_delay
+            .map(humantime::format_duration)
+            .map(tracing::field::display);
+        warn!(
+            parent: self.parent.id(),
+            attempt,
+            wait_duration,
+            error = previous_error as &dyn std::error::Error,
+            "failed executing RPC; retrying after after backoff"
+        );
+        futures::future::ready(())
+    }
+}
+
+fn retry_config()
+-> tryhard::RetryFutureConfig<tryhard::backoff_strategies::ExponentialBackoff, OnRetry> {
+    tryhard::RetryFutureConfig::new(u32::MAX)
+        .exponential_backoff(Duration::from_millis(100))
```

Comment:
Should this be using the new `ExecutionApiRetryStrategy`?
```suggestion
fn retry_config() -> tryhard::RetryFutureConfig<ExecutionApiRetryStrategy, OnRetry> {
    tryhard::RetryFutureConfig::new(u32::MAX)
        .custom_backoff(ExecutionApiRetryStrategy {
            delay: Duration::from_millis(100),
        })
```